tests/Unit/Evolution/Systems/Cce/Actions/Test_ScriObserveInterpolated.cpp

```diff
@@ -0,0 +1,306 @@
+  ActionTesting::next_action<evolution_component>(make_not_null(&runner), 0);
+  ActionTesting::next_action<observation_component>(make_not_null(&runner), 0);
+  runner.set_phase(test_metavariables::Phase::Evolve);
+
+  // generate data that will be well behaved for the interpolation and the
+  // decomposition to modes done in the observation routine
+  UniformCustomDistribution<double> time_dist{-0.5, 0.5};
+
+  const double linear_coefficient = value_dist(gen);
+  const double quadratic_coefficient = value_dist(gen);
+  const size_t vector_size =
+      Spectral::Swsh::number_of_swsh_collocation_points(l_max);
+  const size_t data_points = 30;
+
+  // random vector based on modes
+  // Generate data uniform in r with all angular modes
+  SpinWeighted<ComplexModalVector, -2> generated_modes;
+  generated_modes.data() = make_with_random_values<ComplexModalVector>(
+      make_not_null(&gen), make_not_null(&coefficient_distribution),
+      Spectral::Swsh::size_of_libsharp_coefficient_vector(l_max));
+  for (const auto& mode : Spectral::Swsh::cached_coefficients_metadata(l_max)) {
+    if (mode.l < 2) {
+      generated_modes.data()[mode.transform_of_real_part_offset] = 0.0;
+      generated_modes.data()[mode.transform_of_imag_part_offset] = 0.0;
+    }
+    if (mode.m == 0) {
+      generated_modes.data()[mode.transform_of_real_part_offset] =
+          real(generated_modes.data()[mode.transform_of_real_part_offset]);
+      generated_modes.data()[mode.transform_of_imag_part_offset] =
+          real(generated_modes.data()[mode.transform_of_imag_part_offset]);
+    }
+  }
+  const SpinWeighted<ComplexDataVector, -2> random_vector =
+      Spectral::Swsh::inverse_swsh_transform(l_max, 1, generated_modes);
+
+  for (size_t i = 0; i < 3 * data_points; ++i) {
+    // this will give random times that are nonetheless guaranteed to be
+    // monotonically increasing
+    const DataVector time_vector =
+        i + make_with_random_values<DataVector>(
+                make_not_null(&gen), make_not_null(&time_dist), vector_size);
+    ActionTesting::simple_action<evolution_component,
+                                 SetBoundaryValues<Tags::InertialRetardedTime>>(
+        make_not_null(&runner), 0, time_vector);
+    ActionTesting::simple_action<evolution_component,
+                                 SetBoundaryValues<Tags::News>>(
+        make_not_null(&runner), 0,
+        random_vector.data() * (1.0 + linear_coefficient * (time_vector) +
+                                quadratic_coefficient * square(time_vector)));
+
+    // should put the data we just set into the interpolator
+    ActionTesting::next_action<evolution_component>(make_not_null(&runner), 0);
+    // should process the interpolation and write to file
+    ActionTesting::next_action<evolution_component>(make_not_null(&runner), 0);
+    ActionTesting::next_action<evolution_component>(make_not_null(&runner), 0);
```

Comment:
can you check that the action has deleted the volume data it used to interpolate the data?